doc/classes/EditorImportPlugin.xml

```diff
@@ -234,7 +234,79 @@
+				[codeblocks]
+				[gdscript]
+				var _editor: EditorInterface
+
+				func _import(source_file, save_path, options, r_platform_variants, r_gen_files):
+				    # Make textures folder if required.
+				    var source_file_folder: String = source_file.rsplit("/", true, 1)[0]
+				    var textures_folder: String = "%s/textures" % source_file_folder
+				    DirAccess.make_dir_recursive_absolute(textures_folder)
+
+				    # Simulate image export by copying the source file to the textures folder.
+				    var export_path: String = "%s/icon.png" % textures_folder
+				    var export_image_file: FileAccess = FileAccess.open(export_path, FileAccess.WRITE)
+				    export_image_file.store_buffer(FileAccess.get_file_as_bytes("res://game/my_icon.png"))
+				    export_image_file.close()
+
+				    # Refresh view of exported file.
+				    var editor_file_system: EditorFileSystem = _editor.get_resource_filesystem()
+				    editor_file_system.update_file(textures_folder)
+				    editor_file_system.update_file(export_path)
+
+				    # Append exported file as external resource.
+				    var err: Error = self.append_import_external_resource(export_path)
+				    if err != OK:
+				        return err
+
+				    # Load exported image as a Texture2D.
+				    var export_image_tex: Texture2D = ResourceLoader.load(export_path, "Texture2D", ResourceLoader.CACHE_MODE_REPLACE)
+				    if export_image_tex == null:
+				        return ERR_SCRIPT_FAILED
+
+				    ...
+				[/gdscript]
+				[csharp]
+				public EditorInterface editor;
+
+				public override int _Import(string sourceFile, string savePath, Godot.Collections.Dictionary options, Godot.Collections.Array&lt;string&gt; platformVariants, Godot.Collections.Array&lt;string&gt; genFiles)
+				{
+				    // Make textures folder if required.
+				    string sourceFileFolder = sourceFile.Substring(0, sourceFile.LastIndexOf("/"));
+				    string texturesFolder = $"{sourceFileFolder}/textures";
+				    DirAccess.MakeDirRecursiveAbsolute(texturesFolder);
+
+				    // Simulate image export by copying the source file to the textures folder.
+				    string exportPath = $"{texturesFolder}/icon.png";
+				    FileAccess exportImageFile = FileAccess.Open(exportPath, FileAccess.ModeFlags.Write);
+				    exportImageFile.StoreBuffer(FileAccess.GetFileAsBytes("res://game/my_icon.png"));
+				    exportImageFile.Close();
+
+				    // Refresh view of exported file.
+				    EditorFileSystem editorFileSystem = this.editor.GetResourceFilesystem();
```

Comment:
Remove redundant `this` qualifier.

```suggestion
				    EditorFileSystem editorFileSystem = editor.GetResourceFilesystem();
```

Reply:
it's also worth noting that `editor`/`_editor` field isn't really required (and even currently it isn't shown how and where it could/should be initialized) and one can justcan just do:

```suggestion
				    EditorFileSystem editorFileSystem = EditorInterface.Singleton.GetResourceFilesystem();
```